code/game/objects/items/weapons/shields.dm

```diff
@@ -32,13 +32,68 @@
 /obj/item/shield
 	name = "shield"
 	var/base_block_chance = 50
+	var/max_durability = 500
+	var/durability = 500
+
+/obj/item/shield/proc/breakShield(mob/user)
```

Comment:
Can this be a bit more modular for different materals?

Reply:
Simply reimplement the breakshield to drop other stuff.